adr/2018-06-12_Choose_a_serverless_framework.md

```diff
@@ -0,0 +1,42 @@
+# ADR 1: Choose serverless framework/utility
+
+## Status
+Proposed
+
+## Context
+Lambnik is currently written in Python and uses Chalice as a framework to handle development and deployment to AWS Lambda. This raises issues when porting the tool to 
+Node, however, as Chalice is a Python-specific tool. To that end another framework must be selected.
+
+The main factor influencing this decision is a tool's functional similarity to Chalice as this should ease the translation of the project structure to the new set of technologies. As it stands, Chalice is used to deploy Lambnik and configure it's connection with AWS's API Gateway. Resource provisioning is handled apart from Chalice which is an aspect of the project setup that will be preserved in order to provide for more configurability of data sources in the future. Other factors involved in choosing a framework include:
+* Ability to connect to existing project resources
+* Integration into the Node ecosystem
+* Existence of local development tools for the framework
+* Extent of ongoing project support
+* Developer learning curve
+* Ability to automate more of the project than was handled previously
```

Comment:
Speaking of other factors, I think it's worth including the binary handling considerations, per Matt's comments, in the list.